src/application/t2d/generate_features_and_write_to_disk.py

```diff
@@ -7,22 +7,23 @@
 import random
 import time
 from pathlib import Path
+from typing import Any, Optional, Union
 
 import numpy as np
```

Comment:
Seems like you have rewritten the application script to be a bunch of re-usable chunks (big fan). Perhaps we should move the functions to a seperate file and encourage everyone to use them when they create their features. Should speed up the pipeline for everyone! (Make sure to write a note in the readme)
WIll also make it much easier to keep things updated in case we add new data checks or something else to the pipeline.

Reply:
I'm unsure how re-useable they are. When we move TSFlattener out to its own repo, this is definitely something we should lok at!